### PR TITLE
 Added support for App Key required for calling datadog http apis

### PR DESCRIPTION
**Pull request by smicha**
### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

### CODE REVIEW

**Comment by glinton** [comment]
@smicha if you run `gofmt -w plugins/outputs/datadog` and push that, I believe the tests will run properly.

**Comment by glinton** [comment]
Aren't application keys only required for *reading* metrics? Since this is an output plugin, reading shouldn't be required [source](https://docs.datadoghq.com/api/?lang=python#authentication).

**Comment by smicha** [comment]
I was misled by their public documentation too .  But if you log into datadog and try to look at the spot where they  create app keys , it would clearly tell you that they need an app key now along with api key  to make direct http calls to the metrics api. Their statsd agent doesnt need an app key  though . Just an api key would suffice .

**Comment by smicha** [comment]
It is probably something they added recently .  I have confirmed that an app key is required though by hitting their apis through postman , curl , go -   tests :)

**Comment by smicha** [comment]
@glinton Any idea why this appveyor build is failing. Doesnt seem connected to the files I changed. ?

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,5 @@
 /telegraf.exe
 /telegraf.gz
 /vendor
+.idea/
+
```
Comment (danielnelson) [inline .gitignore]: Revert this file, you can put editor specific ignores in your git excludesfile, something like this:
```
$ git config global core.excludesfile = "/home/smicha/.gitignore"
$ echo ".idea/" >> /home/smicha/.gitignore

File: plugins/outputs/datadog/datadog.go [hunk 2]
```diff
@@ -67,6 +74,7 @@ func (d *Datadog) Connect() error {
 }
 
 func (d *Datadog) Write(metrics []telegraf.Metric) error {
+
```
[the remark this answers is not in the document]
Comment (smicha) [reply plugins/outputs/datadog/datadog.go]: done

**Comment by danielnelson** [comment]
@smicha If the appkey is optional, is there a reason to add it?

**Comment by smicha** [comment]
At this point ,the only reason I can think off is to keep it consistent with their public documentation . Its misleading .  I do have another question though . I tried building    telegraf from source  and tried running it on one of our windows servers . I got an error saying this version of telegraf is not compatible with the version of windows. Is there a way to compile it with some flag that would make it compatible with windows server 2012 r2

**Comment by danielnelson** [comment]
Strange, can you copy me the exact error message?  Are all of the systems 64-bit Windows?

**Comment by smicha** [comment]
screen shot 2018-09-11 at 3 26 02 pm

**Comment by smicha** [comment]
I see a build.py file. Is there an usage document somewhere for it . Can that be used to produce a windows distribution ?

**Comment by smicha** [comment]
ah nevermind. grep failure . i see its used in the  makfile :)

**Comment by danielnelson** [comment]
You can use build.py to make the zip/deb/rpm/etc packages, you can also save some time by only building the platform you are interested in:
```
./scripts/build.py --package --platform=windows --arch=amd64
```

Here is how to build just the binary for another platform:
```
$ GOOS=windows GOARCH=amd64 make telegraf
```

If the appkey is not required then I think we should just skip this change.

**Pull request closed without merging**